ts/features/bonus/bpd/screens/details/transaction/detail/BpdTransactionDetailComponent.tsx

```diff
@@ -60,11 +60,18 @@ const CancelBadge = () => (
 const Table = (props: Props) => (
   <View>
     <View style={styles.rowId}>
-      <H5 weight={"Regular"} color={"bluegrey"}>
-        {I18n.t("payment.details.info.dateAndTime")}
+      <H5 weight={"Regular"} color={"bluegrey"} testID="dateLabel">
+        {(props.transaction.trxDate.getHours() === 0 && props.transaction.trxDate.getMinutes() === 0) ?
+          I18n.t("payment.details.info.onlyDate") : 
+          I18n.t("payment.details.info.dateAndTime")}
       </H5>
-      <H4 weight={"SemiBold"} color={"bluegreyDark"}>
-        {localeDateFormat(
+      <H4 weight={"SemiBold"} color={"bluegreyDark"} testID="dateValue">
+        { (props.transaction.trxDate.getHours() === 0 && props.transaction.trxDate.getMinutes() === 0) ?
```

Comment:
as improvement, could it be better to calculate a const once instead of make the same op twice?

```typescript
const isMidNight =
    props.transaction.trxDate.getHours() +
      props.transaction.trxDate.getMinutes() ===
    0;
```

Reply:
Thanks, your solution is more clear and readable. I'll change as you suggest